### ISSUE TITLE
ld logger compilation issue on SLES12

### ISSUE THREAD

**Issue by ofa5er** [Ericsson/codechecker#669]
I am trying to run CodeChecker on SLES12. However, I am getting a compilation error for ld logger.
I get this error when I do make package:
[INFO] - Getting external dependency projects done.
[INFO] - Removing previous package
[INFO] - Creating package layout
[INFO] - Checking source: llvm scan-build-py (intercept)
[WARNING] - Not exists, build ld logger
[INFO] - Building ld logger ...
[ERROR] - Failed to run: make -f Makefile.manual
[ERROR] - Failed to build ld logger

I run tried to build ld logger manually using make -f Makefile.manual and I got the following errors:

In file included from /usr/include/features.h:402:0,
                 from /usr/include/limits.h:25,
                 from /usr/lib64/gcc/x86_64-suse-linux/4.8/include-fixed/limits.h:168,
                 from /usr/lib64/gcc/x86_64-suse-linux/4.8/include-fixed/syslimits.h:7,
                 from /usr/lib64/gcc/x86_64-suse-linux/4.8/include-fixed/limits.h:34,
                 from src/ldlogger-util.h:12,
                 from src/ldlogger-tool.h:12,
                 from src/ldlogger-tool-gcc.c:9:
/usr/include/gnu/stubs.h:7:27: fatal error: gnu/stubs-32.h: No such file or directory
 # include <gnu/stubs-32.h>

Did any one managed to run CodeChecker on SLES12 ? Do you have a solution to this issue ?

**Comment by bruntib** [comment]
I had a similar issue when compiling the logger, not on SLES but on another system. In my case the error was also related to 32/64 bits, and installing gcc-multilib solved the problem. Make sure that all the dependencies listed in https://github.com/Ericsson/codechecker#install are installed.

**Comment by gyorb** [comment]
One additional information to @bruntib's answer, if you want to build the 64bit version only you can do it with the `make -f Makefile.manual pack64bit` command, after building manually the CodeChecker package builder should detect that the logger is already built.

**Comment by gyorb** [comment]
@ofa5er did you manage to build the logger?

**Comment by gyorb** [comment]
@ofa5er did installing the `gcc-multilib` package solved you problem?

**Comment by ofa5er** [comment]
@gyorb No, it didn't solve the problem. However, I didn't try to build only the 64bit version I managed to find a precompiled version for SLES11 with all the dependencies. It worked fine.

**Comment by Wangxishu** [comment]
I have the same issue in centos7/x64 system

Change -m32 -> -m64 and only build 64bit version of logger
```
# ldlogger executable
ldlogger: $(LDLOGGER_SOURCES) $(LDLOGGER_HEADERS)
	$(CC) -m64 $(CPPFLAGS) $(CFLAGS) -D__LOGGER_MAIN__ $(LDFLAGS) $(LDLOGGER_SOURCES) -o $@
```
also change
scripts/build_package.py
```
    if arch == '32':
        make_cmd = ['make', '-f', 'Makefile.manual', 'pack32bit']
    elif arch == '64':
        make_cmd = ['make', '-f', 'Makefile.manual', 'pack64bit']
    else:
        make_cmd = ['make', '-f', 'Makefile.manual', 'pack64bit']
```

looks like the 'arch' detection has a bug.

then make clean
make package works

**Comment by bruntib** [comment]
I could compile ld-logger on old systems recently, so I would close this ticket. If this issue is still relevant in some environment, then we could reopen it.
